### PR TITLE
Rabby dev

### PR DESCRIPTION
**Pull request by R313mon**
Will remove unused lines of code after all of us have merged with PR3-Dev branch to prevent major merge conflicts with each other.

### CODE REVIEW

File: emuu-client/src/App.css [hunk 1]
```diff
@@ -1,5 +1,5 @@
 .App {
-background: black;
+background: #0f0f0f;
```
Comment (Tanvir-Rahman123) [inline emuu-client/src/App.css:2]: Try making the color a bit lighter, searched it up and it still does look very close to black, but may look different in the application, will have to make a team decision on what looks best.

File: emuu-client/src/components/CreatorsPage/Feeds.js [hunk 2]
```diff
@@ -102,8 +101,10 @@ function Feeds({ setVideo, setUserProfile }) {
             <div className="videos__container">
               {recentVideos &&
                 recentVideos.map((video) => (
-                  <Card sx={{ maxWidth: 380, height: 400 }}>
```
Comment (Tanvir-Rahman123) [inline emuu-client/src/components/CreatorsPage/Feeds.js:105]: Good work on getting the video cards added.

File: emuu-client/src/components/NavbarPostLogin/HeaderPostLogin.js [hunk 3]
```diff
@@ -73,7 +73,7 @@ function HeaderPostLogin({ search, setSearch }) {
           bg="dark"
           variant="dark"
           expand={expand}
-          className="mb-3"
+          className=""
```
Comment (Tanvir-Rahman123) [inline emuu-client/src/components/NavbarPostLogin/HeaderPostLogin.js:76]: If it's an empty className, why not delete the line if it isn't needed?

File: emuu-client/src/components/UserProfile/Feeds.js [hunk 4]
```diff
@@ -219,10 +223,16 @@ function Feeds({ setVideo }) {
                           fontWeight="medium"
                           fontSize="18px"
                         >
-                          {" "}
-                          {video.Username} &ensp;&ensp;&ensp;&ensp;&ensp;
                           {video.Likes} Likes &#x2022; {video.Views} Views
                         </Typography>
+                        <Typography
```
Comment (Tanvir-Rahman123) [inline emuu-client/src/components/UserProfile/Feeds.js:228]: Good work on making creator and profile pages consistent and uniform.

File: emuu-client/src/components/UserProfile/Profile.js [hunk 5]
```diff
@@ -16,29 +18,89 @@ import {
...
       return true;
     return false;
   }
 
-  function uploadBackground(e) {
-    let file = e.target.files[0];
-
-    if (!verifyJpeg(file.name)) return;
+  function uploadBackground(croppedImage) {
+    // let file = e.target.files[0];
```
Comment (Moamin1) [inline emuu-client/src/components/UserProfile/Profile.js:97]: Delete commented code

File: emuu-client/src/components/UserProfile/Profile.js [hunk 6]
```diff
@@ -1,11 +1,13 @@
-import React, { useState, useEffect } from "react";
+import React, { useState, useEffect, useCallback } from "react";
 import AddIcon from "@mui/icons-material/Add";
 import "./Profile.scss";
 import "../../Firebase.js";
 import Feeds from "./Feeds";
 import UserInfo from "./UserInfo";
 import { ref, getStorage, uploadBytes, getDownloadURL } from "firebase/storage";
+import { Blob } from "firebase/firestore";
```
Comment (Tanvir-Rahman123) [inline emuu-client/src/components/UserProfile/Profile.js:8]: Not seeing blob used in the code, get rid of unused import.

File: emuu-client/src/components/UserProfile/Profile.js [hunk 7]
```diff
@@ -16,29 +18,89 @@ import {
...
+  }, []);
+
+  const showCroppedImage = useCallback(async () => {
+    try {
+      const croppedImage = await getCroppedImg(
+        imageSrc,
+        croppedAreaPixels,
+        rotation
+      );
+      console.log("donee", { croppedImage });
```
Comment (Tanvir-Rahman123) [inline emuu-client/src/components/UserProfile/Profile.js:66]: Get rid of console log

File: emuu-client/src/components/UserProfile/Profile.js [hunk 8]
```diff
@@ -16,29 +18,89 @@ import {
...
+  function uploadBackground(croppedImage) {
+    // let file = e.target.files[0];
+    //  if (!verifyJpeg(file.name)) return;
     const storage = getStorage();
-    const storageRef = ref(storage, "/images/" + file.name);
+    const storageRef = ref(storage, "/images/" + uid());
 
     // 'file' comes from the Blob or File API
-    uploadBytes(storageRef, file).then((snapshot) => {
+    uploadString(storageRef, croppedImage, "data_url").then((snapshot) => {
```
Comment (Tanvir-Rahman123) [inline emuu-client/src/components/UserProfile/Profile.js:103]: Are we verifying if it's a jpeg file again? Seems like the code for that is commented out.

File: emuu-client/src/components/UserProfile/Profile.js [hunk 9]
```diff
@@ -90,30 +154,80 @@ function Profile({ setVideo, video }) {
...
+                  </React.Fragment>
+                ) : (
+                  <div className="user-profile-bg-image">
+                    <img id="prf-bg-img" src={Banner} alt="" srcSet="" />
+                    <input
+                      style={{ display: "none" }}
+                      id="background-inp"
+                      type="file"
+                      onChange={(e) => onFileChange(e)}
+                      accept="image/jpeg"
```
Comment (Tanvir-Rahman123) [inline emuu-client/src/components/UserProfile/Profile.js:212]: Disregard jpeg comment. I see that you're only accepting jpeg in here.

File: emuu-client/src/components/UserProfile/canvasUtils.js [hunk 10]
```diff
@@ -0,0 +1,114 @@
...
+  }
+
+  ctx.translate(canvas.width / 2, canvas.height / 2);
+  ctx.rotate((rotation * Math.PI) / 180);
+  ctx.drawImage(image, -image.width / 2, -image.height / 2);
+
+  return new Promise((resolve) => {
+    canvas.toBlob((file) => {
+      resolve(URL.createObjectURL(file));
+    }, "image/png");
```
Comment (Tanvir-Rahman123) [inline emuu-client/src/components/UserProfile/canvasUtils.js:112]: should be jpeg file, not png

File: emuu-client/src/components/UserProfile/Profile.js [hunk 11]
```diff
@@ -150,5 +266,16 @@ function Profile({ setVideo, video }) {
     </>
   );
 }
+function readFile(file) {
+  return new Promise((resolve) => {
```
Comment (Moamin1) [inline emuu-client/src/components/UserProfile/Profile.js:270]: This function should be declared at the top

File: emuu-client/src/components/UserProfile/Profile.js [hunk 12]
```diff
@@ -150,5 +266,16 @@ function Profile({ setVideo, video }) {
...
+    const reader = new FileReader();
+    reader.addEventListener("load", () => resolve(reader.result), false);
+    reader.readAsDataURL(file);
+  });
+}
+
+const StyledDemo = withStyles(styles)(Profile);
 
+const rootElement = document.getElementById("root");
```
Comment (Moamin1) [inline emuu-client/src/components/UserProfile/Profile.js:279]: these consts should be declared at the top as well

File: emuu-client/src/components/home/Home.js [hunk 13]
```diff
@@ -279,10 +279,16 @@ function Home({ setVideo }, { setUserProfile }) {
...
-                                {video.Username} &ensp;&ensp;&ensp;&ensp;&ensp;
                                 {video.Likes} Likes &#x2022; {video.Views} Views
                               </Typography>
+                              <Typography
+                                variant="body2"
+                                color="text.secondary"
+                                fontWeight="medium"
+                                fontSize="18px"
+                              >
+                                {video.Username}
```
Comment (Tanvir-Rahman123) [inline emuu-client/src/components/home/Home.js:290]: Link username to profile page

**Review by Tanvir-Rahman123 — COMMENTED** [review]
Good work for upcoming prototype 3 and for getting react easy crop working. Delete unused functions, and I seen a js file that was used just for styling? Not sure if it's needed, but if it's not, then delete it or use CSS.

**Review by Moamin1 — COMMENTED** [review]
Great job on getting the definite height and width of the banner image cropping and getting the aspect ratio for the image.

**Pull request merged**